c7n/resources/account.py

```diff
@@ -970,3 +970,67 @@ def process(self, resources):
                 if e.response['Error']['Code'] == 'ResourceNotFoundException':
                     return
                 raise
+
+
+@filters.register('xray-encrypt-key')
+class XrayEncrypted(Filter):
+    """Determine if xray is encrypted.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: xray-encrypt-with-default
+                resource: aws.account
+                filters:
+                  - type: xray-encrypt-key
+                    key: default
+              - name: xray-encrypt-with-kms
+                  - type: xray-encrypt-key
+                    key: kms
+    """
+
+    permissions = ('xray:GetEncryptionConfig',)
+    schema = type_schema(
+        'xray-encrypt-key',
+        required=['key'],
+        key={'type': 'string', 'enum': ['default', 'kms']}
+    )
+
+    def process(self, resources, event=None):
+        client = self.manager.session_factory().client('xray')
+        return [{'key': client.get_encryption_config()['EncryptionConfig']['KeyId']}]
```

Comment:
the return value here is still wrong, we should be returning the resources that are filtered in, not the key

Reply:
Can you be more specific? I'm not sure I understand what "returning the resources that are filtered in, not the key" means.

Reply:
all filters return resources, based on the results of what happens in the `process` function. In this case, you would be returning the account resource if the user-defined filter matches the value in the account.

example policy:
```yaml
policies:
  - name: xray-with-kms
    resource: account
    filters:
      - type: xray-encrypt-key
        key: kms
```
the above policy should only return an account resource if the account is using kms encrypted xray. Ideally, we would also return the encryption configuration as an annotated attribute on the resource like so:

```json
{
  "account_id": "123123123123",
  "account_name": "test-account",
  "c7n:XrayEncryptionConfig": {
     ...
   }
}
```

as both filters and actions use the resources filtered in by the previous filters and actions, returning the encryption configuration here would break the downstream filters/actions

Reply:
Ok got it. Here is what is returned...
[
  {
    "account_id": "09.....",
    "account_name": "c-h-dev",
    "c7n:XrayEncryptionConfig": {
      "KeyId": "arn:aws:kms:us-east-1:..foo....",
      "Status": "ACTIVE",
      "Type": "KMS"
    }
  }
]

I believe this is what you expected.